spacy/cli/project/run.py

```diff
@@ -308,3 +330,26 @@ def get_fileinfo(project_dir: Path, paths: List[str]) -> List[Dict[str, Optional
         md5 = get_checksum(file_path) if file_path.exists() else None
         data.append({"path": path, "md5": md5})
     return data
+
+
+def _validate_requirements(projects_path: Path) -> Tuple[List[str], List[str]]:
```

Comment:
I think it would make sense to make this a little more general by working from a list of requirement lines instead of a path + hard-coded filename. `project_run` can handle reading the file for a standard project setup.

I think the messages should only be warnings and I don't think we should ever exit as a result. Someone might want to run an individual step from a project (say a corpus preprocessing step) without having everything installed, and we shouldn't prevent them from doing this.

I think I'd suggest moving the warnings into this method. Maybe it could return a status code instead of lists of packages?